core/src/impl/Kokkos_Profiling.hpp

```diff
@@ -144,6 +145,7 @@ struct DirectFenceIDHandle {
 //
 template <typename Space>
 uint32_t idForInstance(const uintptr_t instance) {
+  const std::lock_guard<std::mutex> lock(std::mutex{});
```

Comment:
Doesn't this create a local variable of type std::mutex that is different for every functional call? Shouldn't this be something like
```suggestion
  static std::mutex instance_mutex;
  const std::lock_guard<std::mutex> lock(instance_mutex);
```

Reply:
Oh right. Slow moment, thanks

Reply:
a51ebe9